File: server/Services/RecipesService.cs


namespace allSpice.Services;


public class RecipesService(RecipesRepository repo)
{
    private readonly RecipesRepository repo = repo;

    internal string DeleteRecipe(int recipeId, string userId)
    {
        Recipe original = GetRecipeById(recipeId);
        if (original.CreatorId != userId) throw new Exception("NOPE");

        original.Deleted = true;
        repo.Update(original);
        return $"deleted {original.Title}";
    }

    internal Recipe CreateRecipe(Recipe recipeData)
    {
        Recipe recipe = repo.Create(recipeData);
        return recipe;
    }

    internal Recipe GetRecipeById(int recipeId)
    {
        Recipe recipe = repo.GetById(recipeId);
        if (recipe == null) throw new Exception($"no recipe at id: {recipeId}");
        return recipe;
    }

    internal List<Recipe> GetAllRecipes()
    {
        List<Recipe> recipes = repo.GetAll();
        return recipes;
    }

    internal Recipe UpdateRecipe(Recipe updateData, string userId)
    {
        Recipe original = GetRecipeById(updateData.Id);
        if (original.CreatorId != userId) throw new Exception("NOPE");

        original.Title ??= updateData.Title;
        original.Img ??= updateData.Img;
        original.Category ??= updateData.Category;
        original.Instructions ??= updateData.CreatorId;

        Recipe update = repo.UpdateRecipe(original);
        return update;
    }
}

File: client/src/services/RecipesService.js

import { AppState } from "../AppState.js"
import { Recipe } from "../models/Recipe.js"
import { api } from "./AxiosService.js"

class RecipesService {
    async getAllRecipes() {
        const response = await api.get('api/recipes')
        console.log('got recipe', response.data)
        AppState.recipes = response.data.map(recipe => new Recipe(recipe))
    }

    // async getRecipeById(recipeId) {
    //     const response = await api.get(`api/recipes/${recipeId}`)
    //     console.log('sup', response.data)
    //     AppState.activeRecipe = new Recipe(response.data)
    // }

    async createRecipe(recipeData) {
        const response = await api.post('api/recipes', recipeData)
        console.log('yeppers', response.data)
        const newRecipe = new Recipe(response.data)
        AppState.recipes.push(newRecipe)
        return newRecipe
    }

    // async deleteRecipe(recipeId) {
    //     const response = await api.delete(`api/recipes/${recipeId}`);
    //     return response.data;
    // }

}

export const recipesService = new RecipesService()